lib/reek/examiner.rb

```diff
@@ -33,12 +33,14 @@ def handle(_exception)
     #   The configuration for this Examiner.
     #
     # @public
+    # :reek:LongParameterList { max_params: 6 }
     def initialize(source,
+                   stdin_filename: nil,
```

Comment:
Since we're in the process of releasing Reek 5, we could also require `source` to be a `Source::SourceCode` and avoid passing in this extra parameter.

Notice that `stdin_filename` is passed to `Source::SourceCode.from` and then read back through `source.origin`, so an alternative would be to *not* pass it to `Source::SourceCode` but store it and use it in `Examiner#description`.